src/lib/azure/azure.ts

```diff
@@ -0,0 +1,179 @@
+export class AzureOpenAI extends OpenAI {
+  private _microsoftEntraTokenProvider: (() => string) | undefined;
+  /**
+   * API Client for interfacing with the Azure OpenAI API.
+   *
+   * @param {string | undefined} [opts.endpoint=process.env['OPENAI_API_VERSION'] ?? undefined]
+   * @param {string | undefined} [opts.apiVersion=process.env['AZURE_OPENAI_ENDPOINT'] ?? undefined] - Your Azure endpoint, including the resource, e.g. `https://example-resource.azure.openai.com/`
+   * @param {string | undefined} [opts.apiKey=process.env['AZURE_OPENAI_API_KEY'] ?? undefined]
+   * @param {string | undefined} opts.deployment - A model deployment, if given, sets the base client URL to include `/deployments/{deployment}`.   * @param {string | null | undefined} [opts.organization=process.env['OPENAI_ORG_ID'] ?? null]
+   * @param {string} [opts.baseURL=process.env['OPENAI_BASE_URL']] - Sets the base URL for the API.
+   * @param {number} [opts.timeout=10 minutes] - The maximum amount of time (in milliseconds) the client will wait for a response before timing out.
+   * @param {number} [opts.httpAgent] - An HTTP agent used to manage HTTP(s) connections.
+   * @param {Core.Fetch} [opts.fetch] - Specify a custom `fetch` function implementation.
+   * @param {number} [opts.maxRetries=2] - The maximum number of times the client will retry a request.
+   * @param {Core.Headers} opts.defaultHeaders - Default headers to include with every request to the API.
+   * @param {Core.DefaultQuery} opts.defaultQuery - Default query parameters to include with every request to the API.
+   * @param {boolean} [opts.dangerouslyAllowBrowser=false] - By default, client-side use of this library is not allowed, as it risks exposing your secret API credentials to attackers.
+   */
+  constructor({
+    baseURL = Core.readEnv('OPENAI_BASE_URL'),
+    apiKey = Core.readEnv('AZURE_OPENAI_API_KEY'),
+    apiVersion = Core.readEnv('OPENAI_API_VERSION'),
+    endpoint,
+    deployment,
+    microsoftEntraTokenProvider,
+    dangerouslyAllowBrowser,
+    ...opts
+  }: AzureClientOptions = {}) {
+    if (apiVersion === undefined) {
+      throw new Errors.OpenAIError(
+        "The OPENAI_API_VERSION environment variable is missing or empty; either provide it, or instantiate the AzureOpenAI client with an apiVersion option, like new AzureOpenAI({ apiVersion: 'My API Version' }).",
+      );
+    }
+
+    if (typeof microsoftEntraTokenProvider === 'function') {
+      dangerouslyAllowBrowser = true;
+    }
+
+    if (!microsoftEntraTokenProvider && !apiKey) {
+      throw new Errors.OpenAIError(
+        'Missing credentials. Please pass one of `apiKey` and `microsoftEntraTokenProvider`, or set the `AZURE_OPENAI_API_KEY` environment variable.',
+      );
+    }
+
+    if (microsoftEntraTokenProvider && apiKey) {
+      throw new Errors.OpenAIError(
+        'The `apiKey` and `microsoftEntraTokenProvider` arguments are mutually exclusive; Only one can be passed at a time.',
```

Comment:
```suggestion
        'The `apiKey` and `microsoftEntraTokenProvider` arguments are mutually exclusive; only one can be passed at a time.',
```

Reply:
Hah, hi Brian! Fan of your work, fun to see you here.